web_notify/models/res_users.py

```diff
@@ -23,24 +25,35 @@ def _compute_channel_names(self):
         compute='_compute_channel_names')
 
     @api.multi
-    def notify_info(self, message, title=None, sticky=False):
+    def notify_info(self, message, title=None, sticky=False, commit=False):
         title = title or _('Information')
         self._notify_channel(
-            'notify_info_channel_name', message, title, sticky)
+            'notify_info_channel_name', message, title, sticky, commit=commit)
 
     @api.multi
-    def notify_warning(self, message, title=None, sticky=False):
+    def notify_warning(self, message, title=None, sticky=False, commit=False):
         title = title or _('Warning')
         self._notify_channel(
-            'notify_warning_channel_name', message, title, sticky)
+            'notify_warning_channel_name', message, title, sticky,
+            commit=commit)
 
     @api.multi
-    def _notify_channel(self, channel_name_field, message, title, sticky):
+    def _notify_channel(self, channel_name_field, message, title, sticky,
+                        commit=False):
         bus_message = {
             'message': message,
             'title': title,
             'sticky': sticky
         }
         notifications = [(getattr(record, channel_name_field), bus_message)
                          for record in self]
-        self.env['bus.bus'].sendmany(notifications)
+        if commit:
+            with odoo.api.Environment.manage():
+                with odoo.registry(self.env.cr.dbname).cursor() as new_cr:
+                    new_env = api.Environment(new_cr, self.env.uid,
+                                              self.env.context)
+                    self.env['bus.bus'].with_env(new_env).sendmany(
+                        notifications)
+                    new_env.cr.commit()
```

Comment:
[This is implicit](https://github.com/odoo/odoo/blob/fc2e80cb4bcc450762c7ac5cb82a3e2d88062b38/odoo/sql_db.py#L391), no need for it.

OTOH, [quoting from guidelines](https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#never-commit-the-transaction):

> All cr.commit() calls outside of the server framework from now on must have an explicit comment explaining why they are absolutely necessary, why they are indeed correct, and why they do not break the transactions. Otherwise they can and will be removed!

So, although you will remove this line and don't have a handwritten call to commit, this `with` block deserves a comment on why is it needed, or this should be rejected according to guidelines.